### PR TITLE
[IMP] web, *: replace search more with view all

### PR DESCRIPTION
**Pull request by juliusc2066**
This commit changes the behavior of the "Search More..." option in Many2XAutocomplete and analytic distribution by making it available as long as there's at least one record in the search results and by renaming it to "View all".

task-3258625

### CODE REVIEW

File: addons/analytic/static/src/components/analytic_distribution/analytic_distribution.js [hunk 1]
```diff
@@ -247,9 +247,9 @@ export class AnalyticDistribution extends Component {
 
         if (searchLimit < records.length) {
```
Comment (h4818) [inline addons/analytic/static/src/components/analytic_distribution/analytic_distribution.js:248]: I guess the "View All" option should always be visible?

**Review by h4818 — COMMENTED** [review]
Thanks for taking care of the analytic widget too

File: addons/web/static/tests/search/search_panel_tests.js [hunk 2]
```diff
@@ -3459,19 +3459,22 @@ QUnit.module("Search", (hooks) => {
...
-    QUnit.test("Don't display empty state message when some filters are availible", async (assert) => {
-        const { TestComponent } = makeTestComponent();
-        await makeWithSearch({
-            serverData,
-            Component: TestComponent,
-            resModel: "partner",
-            searchViewId: false,
-        });
+    QUnit.test(
+        "Don't display empty state message when some filters are availible",
```
Comment (Polymorphe57) [inline addons/web/static/tests/search/search_panel_tests.js:3463]: available

File: addons/web/static/tests/views/fields/many2one_field_tests.js [hunk 3]
```diff
@@ -1156,7 +1156,7 @@ QUnit.module("Fields", (hooks) => {
     QUnit.test("many2one in edit mode", async function (assert) {
         assert.expect(17);
 
-        // create 10 partners to have the 'Search More' option in the autocomplete dropdown
+        // create 10 partners to have the 'View allption in the autocomplete dropdown
```
Comment (Polymorphe57) [inline addons/web/static/tests/views/fields/many2one_field_tests.js:1159]: ?

File: addons/web/static/tests/views/fields/many2one_field_tests.js [hunk 4]
```diff
@@ -1238,11 +1238,11 @@ QUnit.module("Fields", (hooks) => {
             "value of the m2o should have been correctly updated"
         );
 
-        // change the value of the m2o with a record in the 'Search More' modal
+        // change the value of the m2o with a record in the 'View allodal
         await clickDropdown(target, "trululu");
-        // click on 'Search More' (mouseenter required by ui-autocomplete)
+        // click on 'View allmouseenter required by ui-autocomplete)
```
Comment (Polymorphe57) [inline addons/web/static/tests/views/fields/many2one_field_tests.js:1243]: ?

File: addons/web/static/tests/views/fields/many2many_tags_field_tests.js [hunk 5]
```diff
@@ -608,8 +608,8 @@ QUnit.module("Fields", (hooks) => {
         const autocomplete = target.querySelector("[name='timmy'] .o-autocomplete.dropdown");
         assert.strictEqual(
             autocomplete.querySelectorAll("li").length,
-            3,
-            "autocomplete dropdown should have 3 entries (2 values + 'Search and Edit...')"
+            4,
+            "autocomplete dropdown should have 4 entries (2 values + 'View all' + 'Start typing...')"
```
Comment (Polymorphe57) [inline addons/web/static/tests/views/fields/many2many_tags_field_tests.js:612]: why?

File: addons/web/static/tests/views/fields/many2many_tags_field_tests.js [hunk 6]
```diff
@@ -495,12 +495,12 @@ QUnit.module("Fields", (hooks) => {
 
         assert.strictEqual(
             autocompleteDropdown.querySelectorAll("li").length,
-            2,
-            "autocomplete dropdown should have 2 entry"
+            3,
+            "autocomplete dropdown should have 2 entries"
```
Comment (Polymorphe57) [inline addons/web/static/tests/views/fields/many2many_tags_field_tests.js:499]: 3

File: addons/web/static/src/legacy/js/fields/relational_fields.js [hunk 7]
```diff
@@ -681,7 +681,7 @@ var FieldMany2One = AbstractField.extend({
 
         // Add "Search more..." option if results count is higher than the limit
         if (this.limit < values.length) {
```
Comment (Polymorphe57) [inline addons/web/static/src/legacy/js/fields/relational_fields.js:683]: the system do not change there?

File: addons/analytic/static/src/components/analytic_distribution/analytic_distribution.js (hunk 1 above)
Comment (bouvyd) [reply addons/analytic/static/src/components/analytic_distribution/analytic_distribution.js]: Yes, this task is more than a string change :)

the idea is to make it easier for users to open a 'full view' to distinguish records that may be similarly named (and thus the list view allows them to better discriminate between records). So indeed the view all should be shown whenever the dropdown contains *something*

Hope this answers your question!

File: addons/analytic/static/src/components/analytic_distribution/analytic_distribution.js [hunk 8]
```diff
@@ -77,7 +77,7 @@ export class AnalyticDistribution extends Component {
 
         this.selectCreateIsOpen = false;
```
Comment (Polymorphe57) [inline addons/analytic/static/src/components/analytic_distribution/analytic_distribution.js:78]: one2many autocompletes nope: you change Many2XAutocomplete, that means many more

Comment (Polymorphe57) [reply addons/analytic/static/src/components/analytic_distribution/analytic_distribution.js:78]: how many records nope: it is not there if there is no records

**Comment by Polymorphe57** [comment]
robodoo r+

**Pull request closed without merging**